test/e2e/scenario/rum/resources.scenario.ts

```diff
@@ -201,6 +201,33 @@ describe('rum resources', () => {
         expect(resourceEvent?.resource.status_code).toBe(0)
       })
   })
+
+  describe('support XHRs with same XMLHttpRequest instance', () => {
+    createTest('track XHRs when calling requests one after another')
+      .withRum()
+      .withSetup(bundleSetup)
+      .run(async ({ events }) => {
+        await browserExecuteAsync((done) => {
+          const xhr = new XMLHttpRequest()
+          const triggerSecondCall = () => {
+            xhr.removeEventListener('loadend', triggerSecondCall)
+            xhr.addEventListener('loadend', () => done(undefined))
+            xhr.open('GET', '/ok?duration=100&call=2')
+            xhr.send()
+          }
+          xhr.addEventListener('loadend', triggerSecondCall)
+          xhr.open('GET', '/ok?duration=100&call=1')
+          xhr.send()
+        })
+
+        await flushEvents()
+
+        const resourceEvents = events.rumResources.filter((event) => event.resource.type === 'xhr')
+        expect(resourceEvents.length).toEqual(2)
+        expect(events.rumErrors.length).toBe(0)
+        expect(resourceEvents.map((resourceEvent) => resourceEvent.resource.status_code)).toEqual([200, 200])
```

Comment:
Maybe you could test for request URLs (`&call=1`/`&call=2`) to make sure the two resources are independent.

Reply:
Good point